src/aspire/nfft/cufinufft.py

```diff
@@ -0,0 +1,131 @@
+from ctypes import c_int
+
+import logging
+import numpy as np
+import pycuda.autoinit
+import pycuda.driver as cuda
+import pycuda.gpuarray as gpuarray
+from cufinufft import cufinufft
+
+from aspire.nfft import Plan
+from aspire.utils import ensure
+
+logger = logging.getLogger(__name__)
+
+class cuFINufftPlan(Plan):
+    def __init__(self, sz, fourier_pts, epsilon=1e-15, many=0, **kwargs):
+        """
+        A plan for non-uniform FFT (3D)
+
+        :param sz: A tuple indicating the geometry of the signal
+        :param fourier_pts: The points in Fourier space where the Fourier transform is to be calculated,
+            arranged as a 3-by-K array. These need to be in the range [-pi, pi] in each dimension.
```

Comment:
3D only?

Reply:
nice catch in the doc string, thanks. oversight, no, its not just 3d.